Composer/packages/client/src/components/TriggerCreationModal/resolveTriggerWidget.tsx

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import React from 'react';
+import formatMessage from 'format-message';
+import { Label } from 'office-ui-fabric-react/lib/Label';
+import { TextField } from 'office-ui-fabric-react/lib/TextField';
+import { PlaceHolderSectionName } from '@bfc/indexers/lib/utils/luUtil';
+import { UserSettings, DialogInfo, SDKKinds } from '@bfc/shared';
+import { LuEditor, inlineModePlaceholder } from '@bfc/code-editor';
+
+import { TriggerFormData, TriggerFormDataErrors } from '../../utils/dialogUtil';
+import { isRegExRecognizerType, isLUISnQnARecognizerType } from '../../utils/dialogValidator';
+
+import { intentStyles } from './styles';
+import { validateEventName, validateIntentName, getLuDiagnostics, validateRegExPattern } from './validators';
+
+export function resolveTriggerWidget(
+  selectedType: string,
+  dialogFile: DialogInfo | undefined,
+  formData: TriggerFormData,
+  setFormData: (data: TriggerFormData) => void,
+  userSettings: UserSettings,
+  projectId: string,
+  dialogId: string
+) {
+  const isRegEx = isRegExRecognizerType(dialogFile);
+  const isLUISnQnA = isLUISnQnARecognizerType(dialogFile);
+  const showTriggerPhrase = selectedType === SDKKinds.OnIntent && isLUISnQnA;
+
+  const onNameChange = (e, name) => {
```

Comment:
Is it possible to use the `useForm` hook?

Reply:
`useForm` could be used here. During the refactoring, I just moved those code here and I supposed those code would be retired soon since I'm going to break those form logic into different UI widgets and consume them via UI Schema.

Reply:
At that time, I'd like to rewrite these widgets with the `useFrom` hook.